scripts/check_e2e_tests_are_captured_in_ci_test.py

```diff
@@ -0,0 +1,241 @@
+
+    def test_get_e2e_suite_names_from_jobs_travis_yml_file(self):
+        dummy_path = self.swap(
+            check_e2e_tests_are_captured_in_ci, 'read_and_parse_travis_yml_file',
+            self._mock_read_travis_yml_file)
+        with dummy_path:
+            actual_travis_jobs = (
+                check_e2e_tests_are_captured_in_ci
+                .get_e2e_suite_names_from_jobs_travis_yml_file())
+        self.assertEqual(DUMMY_TEST_SUITES, actual_travis_jobs)
+
+    def test_get_e2e_suite_names_from_jobs_travis_yml_file_fail(self):
+        dummy_path = self.swap(
+            check_e2e_tests_are_captured_in_ci, 'read_and_parse_travis_yml_file',
+            self._mock_read_travis_yml_file_fail)
+        with dummy_path:
+            actual_travis_jobs = (
+                check_e2e_tests_are_captured_in_ci
+                .get_e2e_suite_names_from_jobs_travis_yml_file())
+        self.assertNotEqual(DUMMY_TEST_SUITES, actual_travis_jobs)
+
+    def test_get_e2e_suite_names_from_script_travis_yml_file(self):
+        dummy_path = self.swap(
+            check_e2e_tests_are_captured_in_ci, 'read_and_parse_travis_yml_file',
+            self._mock_read_travis_yml_file)
+        with dummy_path:
+            actual_travis_script = (
+                check_e2e_tests_are_captured_in_ci
+                .get_e2e_suite_names_from_script_travis_yml_file())
+        self.assertEqual(DUMMY_TEST_SUITES, actual_travis_script)
+
+    def test_get_e2e_suite_names_from_script_travis_yml_file_fail(self):
+        dummy_path = self.swap(
+            check_e2e_tests_are_captured_in_ci, 'read_and_parse_travis_yml_file',
+            self._mock_read_travis_yml_file_fail)
+        with dummy_path:
+            actual_travis_script = (
+                check_e2e_tests_are_captured_in_ci
+                .get_e2e_suite_names_from_jobs_travis_yml_file())
+        self.assertNotEqual(DUMMY_TEST_SUITES, actual_travis_script)
+
+    def test_get_e2e_suite_names_from_protractor_file(self):
+        dummy_path = self.swap(
+            check_e2e_tests_are_captured_in_ci, 'read_protractor_conf_file',
+            self._mock_read_protractor_conf_file)
+        with dummy_path:
+            actual_protractor_suites = (
+                check_e2e_tests_are_captured_in_ci
+                .get_e2e_suite_names_from_protractor_file())
+        self.assertEqual(DUMMY_TEST_SUITES, actual_protractor_suites)
+
+    def test_get_e2e_suite_names_from_protractor_file_fail(self):
+        dummy_path = self.swap(
+            check_e2e_tests_are_captured_in_ci, 'read_protractor_conf_file',
+            self._mock_read_protractor_conf_file_fail)
+        with dummy_path:
+            actual_protractor_suites = (
+                check_e2e_tests_are_captured_in_ci
+                .get_e2e_suite_names_from_protractor_file())
+        self.assertNotEqual(DUMMY_TEST_SUITES, actual_protractor_suites)
+
+    def test_get_e2e_test_suites_to_exclude_from_travis_ci(self):
```

Comment:
If we are pulling this info to a module-level constant, this test is probably not needed.

Reply:
Yup right. Removed